File: examples/university-demo/src/main/java/org/axonframework/examples/demo/university/faculty/events/CourseRenamed.java
package org.axonframework.examples.demo.university.faculty.events;

import org.axonframework.examples.demo.university.faculty.FacultyTags;
import org.axonframework.examples.demo.university.shared.ids.CourseId;
import org.axonframework.examples.demo.university.shared.ids.FacultyId;
import org.axonframework.eventsourcing.annotation.EventTag;

public record CourseRenamed(
        @EventTag(key = FacultyTags.FACULTY_ID)
        FacultyId facultyId,
        @EventTag(key = FacultyTags.COURSE_ID)
        CourseId courseId,
        String name
) {

}

File: examples/university-demo-kotlin/src/main/kotlin/faculty/events/CourseRenamed.kt
package org.axonframework.examples.university.faculty.events

import org.axonframework.examples.university.faculty.FacultyTags
import org.axonframework.examples.university.shared.ids.CourseId
import org.axonframework.eventsourcing.annotation.EventTag
import org.axonframework.messaging.eventhandling.annotation.Event

@Event
data class CourseRenamed(
    @EventTag(key = FacultyTags.COURSE)
    val courseId: CourseId,
    val name: String
) : FacultyEvent
